libcontainer/mount_linux.go

```diff
@@ -26,8 +27,8 @@ func (e *mountError) Error() string {
 	} else {
 		out += e.target
 	}
-	if e.procfd != "" {
-		out += " (via " + e.procfd + ")"
+	if e.srcFD != "" || e.dstFD != "" {
+		out += " (via " + e.srcFD + ":" + e.dstFD + ")"
```

Comment:
I'm definitely curious to see how this error is shown now.

I wonder how much better this is than a something prints `%+v`, like `fmt.Sprintf("Error: %v trying to: %+v", e.err.Error(), e)`.

I guess probably what we do here is better, though

Reply:
This would look like

```
Error: no such file or directory trying to: {op:mount source:/proc/self/fd/11 srcFD: target:/sys/fs/cgroup/systemd dstFD:/proc/self/fd/12 flags:0x20502f data: err:2}
```
and using the code in this PR looks like
```
mount /proc/self/fd/11:/sys/fs/cgroup/systemd (via :/proc/self/fd/12), flags: 0x20502f: operation not permitted
```

Reply:
I took a fresh look and ended up implementing a different format of an error message, using `src=`, `sfd=`, `dst=`, and `dfd=`, similar to what @thaJeztah suggested.

It makes some error messages look more heavy-weight than they should be, for example:

> umount /some/dir, flags: 0x1: device busy

vs

> umount dst=/some/dir, flags=0x1: device busy

but, I guess, it's even more clear that way